zaza/charm_tests/test_utils.py

```diff
@@ -16,3 +21,136 @@ def _skipIfNotHA_inner_2(*args, **kwargs):
         return _skipIfNotHA_inner_2
 
     return _skipIfNotHA_inner_1
+
+
+class OpenStackAPITest(unittest.TestCase):
+    """Generic helpers for testing OpenStack API charms"""
+
+    @classmethod
+    def setUpClass(cls):
+        cls.keystone_session = openstack_utils.get_overcloud_keystone_session()
+        cls.model_name = lifecycle_utils.get_juju_model()
+        cls.test_config = lifecycle_utils.get_charm_config()
+        cls.application_name = cls.test_config['charm_name']
+
+    def restart_on_changed(self, config_file, default_config, alternate_config,
+                           default_entry, alternate_entry, services):
+        """Test that changing config results in config file being updates and
+           services restarted. Return config to default_config afterwards
+
+        :param config_file: Config file to check for settings
+        :type config_file: str
+        :param default_config: Dict of charm settings to set on completion
+        :type default_config: dict
+        :param alternate_config: Dict of charm settings to change to
+        :type alternate_config: dict
+        :param default_entry: Config file entries that correspond to
+                              default_config
+        :type default_entry: dict
+        :param alternate_entry: Config file entries that correspond to
+        :type alternate_entry: alternate_config
+        :param services: Services expected to be restarted when config_file is
+                         changed.
+        :type services: list
+        """
+        # first_unit is only useed to grab a timestamp, the assumption being
+        # that all the units times are in sync.
+        first_unit = model.get_first_unit_name(
```

Comment:
It would be useful to have this be self.first_unit and maybe established in __init__. There may be several things we will "always" want set.